libs/game/physics.ts

```diff
@@ -215,24 +215,47 @@ class ArcadePhysicsEngine extends PhysicsEngine {
         sprite._lastX = sprite._x;
         sprite._lastY = sprite._y;
 
-        sprite._vx = this.constrain(
-            Fx.add(
-                sprite._vx,
-                Fx.mul(
-                    sprite._ax,
-                    dtSec
+        if (sprite._ax) {
+            sprite._vx = this.constrain(
+                Fx.add(
+                    sprite._vx,
+                    Fx.mul(
+                        sprite._ax,
+                        dtSec
+                    )
                 )
-            )
-        );
-        sprite._vy = this.constrain(
-            Fx.add(
-                sprite._vy,
-                Fx.mul(
-                    sprite._ay,
-                    dtSec
+            );
+        } else if (sprite._fx) {
+            const fx = Fx.mul(sprite._fx, dtSec);
+            const c = Fx.compare(sprite._vx, fx);
+            if (c < 0) // v < f, v = v += f
+                sprite._vx = Fx.add(sprite._vx, fx);
+            else if (c > 0) // v > f, v -= f
+                sprite._vx = Fx.sub(sprite._vx, fx);
+            else
+                sprite._vx = Fx.zeroFx8
+        }
+
+        if (sprite._ay) {
+            sprite._vy = this.constrain(
+                Fx.add(
+                    sprite._vy,
+                    Fx.mul(
+                        sprite._ay,
+                        dtSec
+                    )
                 )
-            )
-        );
+            );
+        } else if( sprite._fy) {
+            const fy = Fx.mul(sprite._fy, dtSec);
+            const c = Fx.compare(sprite._vy, fy);
```

Comment:
Shouldn't this be comparing vy with 0 instead of fy? 

Also, if abs(vy) < abs(fy), this will circle around zero indefinitely, right? Maybe worth a max / min on the results when assigning below to make sure it never passes 0

Reply:
It pulls vy towards 0, assigns it to 0 if abs(vy) < abs(fy)

Reply:
The else handles the close to zero case. Also this bypasses computing vx/vy is ax/ay is zero

Reply:
https://makecode.com/_fosiE8au7YD1 setting fx to 500 causes it to waver between `vx == 500 * dtSec` and `vx == 0` each frame and move to the right

the else here is handling if _vy === fy, so it's the same as if it were omitted and the else if became an else I think

This should work (w/ same for vx)

```typescript
const c = Fx.compare(sprite._vy, Fx.zeroFx8);
if (c < 0)
    sprite._vy = Fx.min(Fx.add(sprite._vy, fy), Fx.zeroFx8);
else if (c > 0)
    sprite._vy = Fx.max(Fx.sub(sprite._vy, fy), Fx.zeroFx8);
```